### PR TITLE
Complete transition away from auto-tests

### PR DESCRIPTION
**Pull request by mikeingold**
## Changes

- Converts all remaining old-style tests (`auto-tests.jl`) to new-style tests (`combinations.jl`), progressing toward completion of #67 
  - [x] `Ball` in 2D
  - [x] `Ball` in 3D
  - [x] `Circle` 
  - [x] `Cylinder`
  - [x] `CylinderSurface`
  - [x] `Disk`
  - [x] `ParaboloidSurface`
  - [x] `Sphere` in 2D
  - [x] `Sphere` in 3D
  - [x] `Tetrahedron`
  - [x] `Torus`
  - [x] `Triangle`
- Reduced CI runtime by about 2 minutes total, addressing #94 

## Future Work/Considerations

This PR will get `combinations.jl` up to around 800 lines. Should this file be partitioned into multiple files? One per geometry type? Separate file for geometries covered by `specializations`?

These new tests replicate the old methodology of using a unit integrand with `Meshes.measure` as a reference solution. More interesting integrand functions and test geometries would be nice in the longer term.

### CODE REVIEW

**Comment by mikeingold** [comment]
I'm temporarily re-marking this as draft. I'm just going to go ahead and finish converting the remaining `auto-tests.jl` content into `combinations.jl` tests with this PR.

File: test/combinations.jl [hunk 1]
```diff
@@ -2,6 +2,56 @@
 # - All supported combinations of integral(f, ::Geometry, ::IntegrationAlgorithm) produce accurate results
 # - Invalid applications of integral aliases (e.g. lineintegral) produce a descriptive error
 
+@testitem "Meshes.Ball 2D" setup=[Setup] begin
+    origin = Point(0, 0)
+    ball = Ball(origin, 2.8)
+
+    f = p -> one(Float64)
```
Comment (JoshuaLampert) [inline test/combinations.jl:9]:
```suggestion
    f(p) = 1.0
```
I think this notation easier to read.

File: test/combinations.jl [hunk 2]
```diff
@@ -2,6 +2,56 @@
...
+    @test_throws "not supported" lineintegral(f, ball)
+    @test surfaceintegral(f, ball) ≈ sol
+    @test_throws "not supported" volumeintegral(f, ball)
+end
+
+@testitem "Meshes.Ball 3D" setup=[Setup] begin
+    origin = Point(0, 0, 0)
+    ball = Ball(origin, 2.8)
+
+    f = p -> one(Float64)
```
Comment (JoshuaLampert) [inline test/combinations.jl:34]:
```suggestion
    f(p) = 1.0
```

File: test/combinations.jl [hunk 3]
```diff
@@ -196,6 +273,85 @@ end
     @test_throws "not supported" volumeintegral(f, cone)
 end
 
+@testitem "Meshes.Cylinder" setup=[Setup] begin
+    pt_w = Point(-1, 0, 0)
+    pt_e = Point(1, 0, 0)
+    cyl = Cylinder(pt_e, pt_w, 2.5)
+
+    f = p -> one(Float64)
```
Comment (JoshuaLampert) [inline test/combinations.jl:281]:
```suggestion
    f(p) = 1.0
```

File: test/combinations.jl [hunk 4]
```diff
@@ -196,6 +273,85 @@ end
...
+    @test_throws "not supported" surfaceintegral(f, cyl)
+    @test volumeintegral(f, cyl) ≈ sol
+end
+
+@testitem "Meshes.CylinderSurface" setup=[Setup] begin
+    pt_w = Point(-1, 0, 0)
+    pt_e = Point(1, 0, 0)
+    cyl = CylinderSurface(pt_e, pt_w, 2.5)
+
+    f = p -> one(Float64)
```
Comment (JoshuaLampert) [inline test/combinations.jl:307]:
```suggestion
    f(p) = 1.0
```

File: test/combinations.jl [hunk 5]
```diff
@@ -265,6 +421,31 @@ end
     @test_throws "not supported" volumeintegral(f, line)
 end
 
+@testitem "Meshes.ParaboloidSurface" setup=[Setup] begin
+    origin = Point(0, 0, 0)
+    parab = ParaboloidSurface(origin, 2.5, 4.15)
+
+    f = p -> one(Float64)
```
Comment (JoshuaLampert) [inline test/combinations.jl:428]:
```suggestion
    f(p) = 1.0
```

File: test/combinations.jl [hunk 6]
```diff
@@ -449,3 +630,134 @@ end
     @test_throws "not supported" surfaceintegral(f, segment)
     @test_throws "not supported" volumeintegral(f, segment)
 end
+
+@testitem "Meshes.Sphere 2D" setup=[Setup] begin
+    origin = Point(0, 0)
+    sphere = Sphere(origin, 4.4)
+
+    f = p -> one(Float64)
```
Comment (JoshuaLampert) [inline test/combinations.jl:638]:
```suggestion
    f(p) = 1.0
```

File: test/combinations.jl [hunk 7]
```diff
@@ -449,3 +630,134 @@ end
...
+    @test lineintegral(f, sphere) ≈ sol
+    @test_throws "not supported" surfaceintegral(f, sphere)
+    @test_throws "not supported" volumeintegral(f, sphere)
+end
+
+@testitem "Meshes.Sphere 3D" setup=[Setup] begin
+    origin = Point(0, 0, 0)
+    sphere = Sphere(origin, 4.4)
+
+    f = p -> one(Float64)
```
Comment (JoshuaLampert) [inline test/combinations.jl:663]:
```suggestion
    f(p) = 1.0
```

File: test/combinations.jl [hunk 8]
```diff
@@ -449,3 +630,134 @@ end
...
+end
+
+@testitem "Meshes.Tetrahedron" setup=[Setup] begin
+    pt_n = Point(0, 1, 0)
+    pt_w = Point(-1, 0, 0)
+    pt_e = Point(1, 0, 0)
+    ẑ = Vec(0, 0, 1)
+    tetrahedron = Tetrahedron(pt_n, pt_w, pt_e, pt_n + ẑ)
+
+    f = p -> one(Float64)
```
Comment (JoshuaLampert) [inline test/combinations.jl:691]:
```suggestion
    f(p) = 1.0
```

File: test/combinations.jl [hunk 9]
```diff
@@ -449,3 +630,134 @@ end
...
+    @test_throws "not supported" surfaceintegral(f, tetrahedron)
+    @test volumeintegral(f, tetrahedron, GaussKronrod()) ≈ sol
+end
+
+@testitem "Meshes.Torus" setup=[Setup] begin
+    origin = Point(0, 0, 0)
+    ẑ = Vec(0, 0, 1)
+    torus = Torus(origin, ẑ, 3.5, 1.25)
+
+    f = p -> one(Float64)
```
Comment (JoshuaLampert) [inline test/combinations.jl:717]:
```suggestion
    f(p) = 1.0
```

File: test/combinations.jl [hunk 10]
```diff
@@ -449,3 +630,134 @@ end
...
+    @test_throws "not supported" volumeintegral(f, torus)
+end
+
+@testitem "Meshes.Triangle" setup=[Setup] begin
+    pt_n = Point(0, 1, 0)
+    pt_w = Point(-1, 0, 0)
+    pt_e = Point(1, 0, 0)
+    triangle = Ngon(pt_e, pt_n, pt_w)
+
+    f = p -> one(Float64)
```
Comment (JoshuaLampert) [inline test/combinations.jl:744]:
```suggestion
    f(p) = 1.0
```

File: test/combinations.jl [hunk 11]
```diff
@@ -449,3 +630,134 @@ end
...
+    @test_throws "not supported" lineintegral(f, torus)
+    @test surfaceintegral(f, torus) ≈ sol
+    @test_throws "not supported" volumeintegral(f, torus)
+end
+
+@testitem "Meshes.Triangle" setup=[Setup] begin
+    pt_n = Point(0, 1, 0)
+    pt_w = Point(-1, 0, 0)
+    pt_e = Point(1, 0, 0)
+    triangle = Ngon(pt_e, pt_n, pt_w)
```
Comment (JoshuaLampert) [inline test/combinations.jl:742]:
```suggestion
    triangle = Triangle(pt_e, pt_n, pt_w)
```
To clarify that this really is a triangle.

**Comment by JoshuaLampert** [comment]
This looks pretty good. In my opinion, we can keep one file for now. The file auto_tests.jl can be deleted now, can't it? Testing more interesting integrand functions would definitely be good.

**Comment by mikeingold** [comment]
> This looks pretty good. In my opinion, we can keep one file for now.

Thanks! Part of me prefers the reduced code redundancy of the auto-test system, but this longer form is also much less complex to debug, and will allow for individualized rule settings and non-unit integrands for everything with time. For now, I didn’t want the lack of analytic solutions to keep holding up the transition.

> The file auto_tests.jl can be deleted now, can't it?

Yes. I was planning to delete it once we’re comfortable that all CI is passing and that those tests are officially dead.

**Comment by JoshuaLampert** [comment]
>Yes. I was planning to delete it once we’re comfortable that all CI is passing and that those tests are officially dead.

Tests look good.

**Comment by mikeingold** [comment]
I'm away from my PC right now so can't `git rm` until I get back. Otherwise I think this one is good to go.

**Comment by JoshuaLampert** [comment]
I removed the file in [2fb6d2b](https://github.com/mikeingold/MeshIntegrals.jl/pull/100/commits/2fb6d2b2ca9551cbdd2b5a3e6a77a0f913c8b262).

**Review by JoshuaLampert — APPROVED** [review]

**Pull request merged**